### PR TITLE
Tooltips for select dropdown

### PR DESCRIPTION
**Pull request by reedsa**
## Merge Checklist

- [X] Has descriptive PR title and body
- [X] All changes are backwards compatible
- [x] QA'd by the SendGrid Style Guide designer
- [x] Code reviewed by a SendGrid frontend lead

Adds tooltip icon and tooltip content with configurable position to select dropdown fields.

### CODE REVIEW

File: src/select/index.tsx [hunk 1]
```diff
@@ -213,7 +215,32 @@ const SelectStyles = {
     };
   },
 };
+
+const DropdownIndicator = (props: any) => {
```
Comment (alexabreu) [inline src/select/index.tsx]: Is there a React Select type we can leverage?

File: src/select/index.tsx [hunk 2]
```diff
@@ -213,7 +215,32 @@ const SelectStyles = {
...
+  if (!props.selectProps.tooltip) {
+    return <components.DropdownIndicator {...props} />;
+  }
+
+  return (
+    components.DropdownIndicator && (
```
Comment (alexabreu) [inline src/select/index.tsx]: do we need this check for the existence of the components.DropdownIndicator

Comment (reedsa) [reply src/select/index.tsx]: This was from their documentation, not sure why. It works great without it so I'll remove!

**Comment by Ditofry** [comment]
I will also bring this into #311 once it is merged

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**